src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -1010,10 +1010,12 @@ private static void NarrowFourUtf16CharsToAsciiAndWriteToBuffer(ref byte outputB
         {
             Debug.Assert(AllCharsInUInt64AreAscii(value));
 
-            if (Bmi2.X64.IsSupported)
+            if (Sse2.X64.IsSupported)
             {
-                // BMI2 will work regardless of the processor's endianness.
-                Unsafe.WriteUnaligned(ref outputBuffer, (uint)Bmi2.X64.ParallelBitExtract(value, 0x00FF00FF_00FF00FFul));
+                Debug.Assert(BitConverter.IsLittleEndian, "SSE2 narrowing assumes little-endian.");
+                Vector128<short> vecWide = Sse2.X64.ConvertScalarToVector128UInt64(value).AsInt16();
```

Comment:
I think it's worth a small comment explaining "why" this works. Intrinsic code is specialized and its not always immediately obvious why a given snippet of code is valid, especially to newcomers to the framework. For example: newcomers may also not be familiar with "saturation" (which clamps to `MaxValue` rather than overflowing).